gap/base/methods.gd

```diff
@@ -0,0 +1,130 @@
+#############################################################################
+##
+##  This file is part of recog, a package for the GAP computer algebra system
+##  which provides a collection of methods for the constructive recognition
+##  of groups.
+##
+##  This files's authors include Sergio Siccha.
+##
+##  Copyright of recog belongs to its developers whose names are too numerous
+##  to list here. Please refer to the COPYRIGHT file for details.
+##
+##  SPDX-License-Identifier: GPL-3.0-or-later
+##
+##
+##  Declarations of recog methods
+##
+#############################################################################
+
+# <#GAPDoc Label="IsRecogMethod">
+# <ManSection>
+# <Filt Name="IsRecogMethod" Type="Category"/>
+# <Description>
+# The category of recognition methods, that is of the objects created via
+# <Ref Func="RecogMethod"/>.
+# </Description>
+# </ManSection>
+# <#/GAPDoc>
+DeclareCategory("IsRecogMethod", IsObject);
+BindGlobal("RecogMethodFamily", NewFamily("RecogMethodFamily", IsRecogMethod));
+BindGlobal("RecogMethodType",
+           NewType(RecogMethodFamily,
+                   IsRecogMethod and IsAttributeStoringRep));
+
+# <#GAPDoc Label="RecogMethod">
+# <ManSection>
+# <Func Name="RecogMethod" Arg="stamp, comment, func"/>
+# <Description>
+# Return a recognition method <C>method</C> in the filter
```

Comment:
Maybe add a comment here, that `RecogMethod` should not be called directly but rather be created via `BindRecogMethod`.